File: python/S14_BST/L1_CONCEPT/1_bst.py
class Node:
    def __init__(self,data):
        self.left=None
        self.right=None
        self.data=data
class bst:
    def __init__(self):
        self.root=None
    def insertbst(self,root,data):
        if root is None:
            return Node(data)
        if root.data>data:
            root.left=self.insertbst(root.left,data)
        elif root.data<data:
            root.right=self.insertbst(root.right,data)
        return root
        
            
            
    def postorder(self,root):
        if root is None:
            return
        self.postorder(root.left)
        self.postorder(root.right)
        print(root.data,end=" ")
if __name__=="__main__":
    b=bst()
    b.root=b.insertbst(b.root,34)
    b.root=b.insertbst(b.root,30)
    b.root=b.insertbst(b.root,35)
    b.root=b.insertbst(b.root,33)
    b.root=b.insertbst(b.root,32)
    b.postorder(b.root)
    

File: JAVA/S14_BST/L1_CONCEPT/_1_bst.java
class Node{
    int data;
    Node left;
    Node right;
    Node(int data){
        this.data=data;
        left=null;
        right=null;
    }
}
class BST{
    Node root;
    Node insertbst(Node root,int data){
        Node newnode=new Node(data);
        if(root==null){
            // Node newnode=new Node(data);
            root=newnode;
            return root;
        }
        if(root.data>data){
            root.left=insertbst(root.left,data);
        }
        else root.right=insertbst(root.right,data);
        return root;

    }
    void postorder(Node root){
        if(root==null)return;
        postorder(root.left);
        postorder(root.right);
        System.out.print(root.data+" ");
    }
}
public class _1_bst{
    public static void main(String[] args){
        BST b=new BST();
        b.root=b.insertbst(b.root,67);
        b.postorder(b.root);
    }
}